crash if generate block has no if_exists attribute

## Describe the bug

terragrunt parser crashes if a generate block is missing if_exists attribute

## Steps To Reproduce

In an empty directory, create a `terragrunt.hcl` file with this content:
```hcl
generate "test" {
  path     = "test.tf"
  contents = "foo"
}
```

run "terragrunt validate". 

terragrunt crashes with:

```
$ terragrunt validate
13:02:34.075 ERROR  Error: Missing required argument
13:02:34.075 ERROR    on ./terragrunt.hcl line 1, in generate "test":
13:02:34.075 ERROR     1: generate "test" {
13:02:34.075 ERROR  The argument "if_exists" is required, but no definition was found.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e23c77]

goroutine 51 [running]:
github.com/gruntwork-io/terragrunt/config.ParseConfig(0xc0005a4300, {0x34869d0, 0xc000cb5248}, 0xc000404180, 0x0)
        /home/circleci/project/config/config.go:1353 +0x737
github.com/gruntwork-io/terragrunt/config.ParseConfigFile.func1({0xc000eef1f8?, 0xc000a3e000?})
        /home/circleci/project/config/config.go:1222 +0x4b3
github.com/gruntwork-io/terragrunt/telemetry.(*Meter).Time(0x48eb8a?, {0x3457268?, 0xc0005a4300?}, {0x2c8ea52?, 0xc000eef3f0?}, 0x28381e0?, 0x264c900?)
        /home/circleci/project/telemetry/meter.go:76 +0x23c
github.com/gruntwork-io/terragrunt/config.ParseConfigFile.(*Telemeter).Collect.func2({0x3457268?, 0xc0005a4300?})
        /home/circleci/project/telemetry/telemeter.go:59 +0x3e
github.com/gruntwork-io/terragrunt/telemetry.(*Tracer).Trace(0x28381e0?, {0x3457268?, 0xc0005a4300?}, {0x2c8ea52?, 0x34b1a20?}, 0xc000404120?, 0xc000eef2a8?)
        /home/circleci/project/telemetry/tracer.go:182 +0x155
github.com/gruntwork-io/terragrunt/telemetry.(*Telemeter).Collect(...)
        /home/circleci/project/telemetry/telemeter.go:58
github.com/gruntwork-io/terragrunt/config.ParseConfigFile(0xc0005a4300, {0x34869d0, 0xc000cb5248}, {0xc0001b8370, 0x4b}, 0x0)
        /home/circleci/project/config/config.go:1175 +0x318
github.com/gruntwork-io/terragrunt/config.ReadTerragruntConfig({0x3456938, 0xc000c584b0}, {0x34869d0, 0xc000cb5248}, 0xc000c2e488, {0xc000404060, 0x3, 0x4})
        /home/circleci/project/config/config.go:1165 +0x1d3
github.com/gruntwork-io/terragrunt/cli/commands/run.run({0x3456938, 0xc000c584b0}, {0x34869d0, 0xc000ce7698}, 0xc000c2e488, 0xc0004a22a0, 0xc000b23a00)
        /home/circleci/project/cli/commands/run/run.go:114 +0x27b
github.com/gruntwork-io/terragrunt/cli/commands/run.Run({0x3456938?, 0xc000c584b0?}, {0x34869d0?, 0xc000ce7698?}, 0x0?, 0x0?)
        /home/circleci/project/cli/commands/run/run.go:91 +0x65
github.com/gruntwork-io/terragrunt/cli/commands/run.NewSubcommands.func1.Action.1(...)
        /home/circleci/project/cli/commands/run/cli.go:90
github.com/gruntwork-io/terragrunt/cli/commands/run.NewSubcommands.func1(0xc0007126e0)
        /home/circleci/project/cli/commands/run/cli.go:69 +0x29d
github.com/gruntwork-io/terragrunt/cli/commands/common/runall.WrapCommand.func1(0x0?, 0x2000000000000?)
        /home/circleci/project/cli/commands/common/runall/cli.go:57 +0x12b
github.com/gruntwork-io/terragrunt/internal/cli.(*Command).WrapAction.func1(0x10?)
        /home/circleci/project/internal/cli/command.go:306 +0x1b
github.com/gruntwork-io/terragrunt/cli/commands/common/graph.WrapCommand.func1(0x264c580?, 0x1?)
        /home/circleci/project/cli/commands/common/graph/cli.go:52 +0x12b
github.com/gruntwork-io/terragrunt/internal/cli.(*Command).WrapAction.func1(0xc000ce7698?)
        /home/circleci/project/internal/cli/command.go:306 +0x1b
github.com/gruntwork-io/terragrunt/cli/commands/run.NewCommand.wrapWithStackGenerate.func2(0xc0007126e0, 0xc000d14930)
        /home/circleci/project/cli/commands/run/cli.go:123 +0x684
github.com/gruntwork-io/terragrunt/internal/cli.(*Command).WrapAction.func1(0x0?)
        /home/circleci/project/internal/cli/command.go:306 +0x1b
github.com/gruntwork-io/terragrunt/cli/commands.runAction.func2()
        /home/circleci/project/cli/commands/commands.go:185 +0x47
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/circleci/go/pkg/mod/golang.org/x/sync@v0.16.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /home/circleci/go/pkg/mod/golang.org/x/sync@v0.16.0/errgroup/errgroup.go:78 +0x95
```

## Expected behavior

terragrunt complains about missing attribute.


## Must haves

- [x] Steps for reproduction provided.

## Nice to haves

- [x] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: v0.86.2
- OpenTofu/Terraform version: OpenTofu v1.10.6 
- Environment details (Ubuntu 20.04, Windows 10, etc.): Debian sid

## Additional context

Note that this bug also impacts `terragrunt-ls` which has the same crash. Which is more puzzling as the missing attribute is not reported before crashing.

--- a/config/config.go
+++ b/config/config.go
@@ -1335,6 +1335,13 @@ func ParseConfig(
 			errs = errs.Append(err)
 			return config, errs.ErrorOrNil()
 		}
+
+		// We should never get a nil config here, so if we do, return the config we've been able to parse so far
+		// and return any errors that have occurred so far to avoid a nil pointer dereference below.
+		if mergedConfig == nil {
+			return config, errs.ErrorOrNil()
+		}
+
 		// Saving processed includes into configuration, direct assignment since nested includes aren't supported
 		mergedConfig.ProcessedIncludes = ctx.TrackInclude.CurrentMap
 		// Make sure the top level information that is not automatically merged in is captured on the merged config to
@@ -1732,9 +1739,16 @@ func convertToTerragruntConfig(ctx *ParsingContext, configPath string, terragrun
 	}
 
 	for _, block := range generateBlocks {
+		// Validate that if_exists is provided (required attribute)
+		if block.IfExists == "" {
+			errs = errs.Append(errors.Errorf("generate block %q is missing required attribute \"if_exists\"", block.Name))
+			continue
+		}
+
 		ifExists, err := codegen.GenerateConfigExistsFromString(block.IfExists)
 		if err != nil {
-			return nil, err
+			errs = errs.Append(errors.Errorf("generate block %q: %w", block.Name, err))
+			continue
 		}
 
 		if block.IfDisabled == nil {

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -1549,6 +1549,27 @@ func TestBestEffortParseConfigString(t *testing.T) {
 	}
 }
 
+func TestParseConfigWithMissingIfExists(t *testing.T) {
+	t.Parallel()
+
+	cfg := `generate "test" {
+  path     = "test.tf"
+  contents = "foo"
+}`
+
+	l := createLogger()
+	ctx := config.NewParsingContext(t.Context(), l, mockOptionsForTest(t))
+
+	terragruntConfig, err := config.ParseConfigString(ctx, l, config.DefaultTerragruntConfigPath, cfg, nil)
+	require.Error(t, err)
+
+	errStr := err.Error()
+	hasIfExistsError := strings.Contains(errStr, "if_exists")
+	hasGenerateError := strings.Contains(errStr, "generate") || strings.Contains(errStr, "Missing required argument")
+	assert.True(t, hasIfExistsError || hasGenerateError, "Error message should mention missing if_exists attribute or generate block. Got: %s", errStr)
+	assert.NotNil(t, terragruntConfig)
+}
+
 func TestBestEffortParseConfigStringWDependency(t *testing.T) {
 	t.Parallel()
 

